src/components/pages/Turbo/modals/CreateSafeModal/index.tsx

```diff
@@ -0,0 +1,148 @@
+    "isOpen" | "onClose"
+>;
+
+export const CreateSafeModal: React.FC<CreateSafeModalProps> = ({
+    isOpen,
+    onClose,
+}) => {
+
+    const { address, provider, chainId } = useRari()
+
+    const [stepIndex, setStepIndex] = useState(0);
+    function incrementStepIndex() {
+        if (stepIndex + 1 !== MODAL_STEPS.length) {
+            setStepIndex(stepIndex + 1);
+        }
+    }
+
+    const [amount, setAmount] = useState<string>("10");
+    const [underlyingToken, setUnderlyingToken] = useState<string | undefined>()
+
+
+    const [creating, setCreating] = useState(false)
+    const tokenData = useTokenData(underlyingToken)
+
+
+    // Utils
+    const toast = useToast();
+
+    const handleCreateSafe = async () => {
+        if (!address || !provider || !chainId || !underlyingToken) return;
+
+
+        console.log({amount})
+
+        const amountBN = parseEther(!!amount ? amount : "0");
+
+        if (!amountBN.isZero()) {
+            try {
+                setCreating(true)
+                const receipt = await createSafeAndDeposit(
+                    provider.getSigner(),
+                    amountBN,
+                    chainId
+                );
+                console.log({ receipt })
+            } catch (err) {
+                handleGenericError(err, toast)
+                console.log({ err })
+            } finally {
+                setCreating(false)
+            }
+        } else {
+            try {
+                alert("createSafe")
+                setCreating(true)
+                const receipt = await createSafe(
+                    underlyingToken,
+                    provider,
+                    chainId
+                )
+                console.log({ receipt })
+            } catch (err) {
+                handleGenericError(err, toast)
+                console.log({ err })
+            } finally {
+                setCreating(false)
+                setStepIndex(0)
+                onClose()
+            }
+        }
+    };
+
+    console.log({ stepIndex })
+
+    return (
+        <Modal
+            isOpen={isOpen}
+            onClose={onClose}
+            title={MODAL_STEPS[stepIndex].title}
+            subtitle={MODAL_STEPS[stepIndex].subtitle}
+            progressValue={(stepIndex + 1) / MODAL_STEPS.length * 100}
+        >
+            {/* {MODAL_STEPS[stepIndex].children} */}
+            {stepIndex === 0 && <CreateSafeStepOne incrementStepIndex={incrementStepIndex} />}
+            {stepIndex === 1 && <CreateSafeStepTwo incrementStepIndex={incrementStepIndex} setUnderlyingToken={setUnderlyingToken} />}
+            {stepIndex === 2 && <CreateSafeStepThree
+                incrementStepIndex={incrementStepIndex}
+                setUnderlyingToken={setUnderlyingToken}
+                underlyingToken={underlyingToken}
+                amount={amount}
+                setAmount={setAmount}
+                handleCreateSafe={handleCreateSafe}
+            />}
```

Comment:
One thing that might be hard to keep track of here is which props are required by which steps (e.g. step 3 requires a bunch more steps that step 2 doesn't necessarily require). I wonder if using context might make more sense instead — one advantage of the context approach is that we may be able to create a `incrementStepIndex` function that doesn't have to be recreated every time `stepIndex` changes (this post on prop drilling is also sort of relevant: https://kentcdodds.com/blog/prop-drilling)